zson/marshal.go

```diff
@@ -597,6 +612,12 @@ func (u *UnmarshalZNGContext) decodeAny(zv zng.Value, v reflect.Value) error {
 			return pv.Interface().(ZNGUnmarshaler).UnmarshalZNG(u, zv)
 		}
 	}
+	if _, ok := v.Interface().(zng.Value); ok {
+		// For embedded zng.Values we simply set the
+		// reflect value to the zng.Value that has been decoded.
+		v.Set(reflect.ValueOf(zv))
+		return nil
+	}
 	if _, ok := v.Interface().(nano.Ts); ok {
```

Comment:
Nit: Move the `zng.Value` if statement after `nano.Ts` one for consistency with `encodeAny`.